kolibri/plugins/coach/assets/src/views/ClassLearnersListPage.vue

```diff
@@ -0,0 +1,214 @@
+<template>
+
+  <CoreBase
+    :immersivePage="true"
+    immersivePageIcon="back"
+    :immersivePagePrimary="false"
+    :immersivePageRoute="$router.getRoute('HomePage')"
+    :appBarTitle="$store.state.classSummary.name"
+  >
+    <KPageContainer>
+      <h1>{{ $tr('pageHeader', { className: $store.state.classSummary.name }) }} </h1>
+      <KButton
+        :text="$tr('howToTroubleshootModalHeader')"
+        appearance="basic-link"
+        class="troubleshooting-modal-link"
+        @click="displayTroubleshootModal = true"
+      />
+      <KModal
+        v-if="displayTroubleshootModal"
+        :title="$tr('howToTroubleshootModalHeader')"
+        size="medium"
+        :submitText="$tr('close')"
+        @submit="displayTroubleshootModal = false"
+      >
+        <div v-for="status in syncStatusOptions" :key="status.id" class="status-option-display">
+          <SyncStatusDisplay :syncStatus="status" displaySize="sync-status-large-bold" />
+          <SyncStatusDescription :syncStatus="status" />
+        </div>
+      </KModal>
+      <CoreTable>
+        <template #headers>
+          <th>
+            {{ coreString('fullNameLabel') }}
+          </th>
+          <th>{{ coreString('usernameLabel') }}</th>
+          <th>{{ $tr('deviceStatus') }}</th>
+          <th>{{ $tr('lastSyncedStatus') }}</th>
+        </template>
+
+        <template #tbody>
+          <tbody>
+            <tr
+              v-for="learner in learnerMap"
+              :key="learner.id"
+            >
+              <td>
+                <KLabeledIcon
+                  icon="person"
+                  :label="learner.name"
+                />
+              </td>
+              <td>
+                <span dir="auto">
+                  {{ learner.username }}
+                </span>
+              </td>
+              <td>
+                <SyncStatusDisplay
+                  :syncStatus="mapSyncStatusOptionToLearner(learner.id)"
+                  displaySize="sync-status-large"
+                />
+              </td>
+              <td>
+                <span dir="auto">
+                  {{ mapLastSynctedTimeToLearner(learner.id) }}
+                </span>
+              </td>
+            </tr>
+          </tbody>
+        </template>
+      </CoreTable>
+    </KPageContainer>
+  </CoreBase>
+
+</template>
+
+
+<script>
+
+  import CoreBase from 'kolibri.coreVue.components.CoreBase';
+  import CoreTable from 'kolibri.coreVue.components.CoreTable';
+  import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
+  import { SyncStatus } from 'kolibri.coreVue.vuex.constants';
+  import { mapState, mapActions } from 'vuex';
+  import SyncStatusDisplay from '../../../../../core/assets/src/views/SyncStatusDisplay';
```

Comment:
One slight issue with doing this import is that we will either duplicate the strings across the core and the coach plugin or the strings will not be included in the coach plugin (I have completely forgotten at this point which way the i18n machinery goes in this regard).

It might be _simpler_ (for a certain value of simple) to add these components to the core API and reference them in the same way as CoreBase and CoreTable.